src/cascadia/TerminalControl/TermControlUiaProvider.cpp

```diff
@@ -82,49 +87,73 @@ IFACEMETHODIMP TermControlUiaProvider::get_FragmentRoot(_COM_Outptr_result_maybe
 
 const COORD TermControlUiaProvider::GetFontSize() const
 {
-    return _termControl.GetActualFont().GetSize();
+    return _termControl->GetActualFont().GetSize();
 }
 
 const winrt::Windows::UI::Xaml::Thickness TermControlUiaProvider::GetPadding() const
 {
-    return _termControl.GetPadding();
+    return _termControl->GetPadding();
 }
 
-std::deque<UiaTextRangeBase*> TermControlUiaProvider::GetSelectionRanges(_In_ IRawElementProviderSimple* const pProvider)
+HRESULT TermControlUiaProvider::GetSelectionRanges(_In_ IRawElementProviderSimple* pProvider, _Out_ std::deque<ComPtr<UiaTextRangeBase>>& result)
+try
 {
-    std::deque<UiaTextRangeBase*> result;
+    typename std::remove_reference<decltype(result)>::type temporaryResult;

```

Comment:
Sorry, what wizardry is this?

Reply:
'Tis a spell from Grandmaester @DHowett-MSFT. IIRC it's just a temporary var for our out param. Then I swap it later. He can confirm.